cpp/include/rapidsmpf/buffer/buffer.hpp

```diff
@@ -37,6 +42,46 @@ class Buffer {
     friend class BufferResource;
 
   public:
+    /**
+     * @brief CUDA event to provide synchronization among set of chunks.
+     *
+     * This event is used to serve as a synchronization point for a set of chunks
+     * given a user-specified stream.
+     */
```

Comment:
It would be good with an enumeration of each case that needs to use events and why.